vtex/affiliates
# Feature/set custom data on order form context

#### What is the purpose of this pull request?

- setAffiliateOnOrderForm mutation to respond with the mutated orderform
- AffiliateValidator component to update OrderFormContext with customData after mutation

#### What problem is this solving?

We want the affiliation of the orderform to be exposed on the client-side, so that the information can be used in any way the client account needs it to be used, such as conditioning marketing tags using Google Tag Manager.

This changes, along with changes made on the [vtex.checkout-graphql](https://github.com/vtex-apps/checkout-graphql/pull/165) and [vtex.checkout-resources](https://github.com/vtex-apps/checkout-resources/pull/90) will make so as the affiliateId present in customData will always be reflected on the OrderFormContext, on the client-side.

#### How to test it?

This [affiliate store](https://gabiru--sandboxbrdev.myvtex.com/affiliates/fulano) inserts the affiliateId on customData. If you check the orderform present on localStorage, you will see that this change is reflected on it.

#### How does this PR make you feel? [:link:](http://giphy.com/)



![](https://media.giphy.com/media/df0cUZrXQ8jt3HpoTK/giphy.gif)

#### Types of changes
